x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateExtractTests.java

```diff
@@ -62,7 +62,31 @@ public static Iterable<Object[]> parameters() {
                             new TestCaseSupplier.TypedData(new BytesRef("YeAr"), DataType.TEXT, "chrono"),
                             new TestCaseSupplier.TypedData(1687944333000L, DataType.DATETIME, "date")
                         ),
-                        "DateExtractEvaluator[value=Attribute[channel=1], chronoField=Attribute[channel=0], zone=Z]",
+                        "DateExtractMillisEvaluator[value=Attribute[channel=1], chronoField=Attribute[channel=0], zone=Z]",
+                        DataType.LONG,
+                        equalTo(2023L)
+                    )
+                ),
+                new TestCaseSupplier(
+                    List.of(DataType.KEYWORD, DataType.DATE_NANOS),
+                    () -> new TestCaseSupplier.TestCase(
+                        List.of(
+                            new TestCaseSupplier.TypedData(new BytesRef("YeAr"), DataType.KEYWORD, "chrono"),
+                            new TestCaseSupplier.TypedData(1687944333000000000L, DataType.DATE_NANOS, "date")
+                        ),
+                        "DateExtractNanosEvaluator[value=Attribute[channel=1], chronoField=Attribute[channel=0], zone=Z]",
+                        DataType.LONG,
+                        equalTo(2023L)
+                    )
+                ),
+                new TestCaseSupplier(
+                    List.of(DataType.TEXT, DataType.DATE_NANOS),
+                    () -> new TestCaseSupplier.TestCase(
+                        List.of(
+                            new TestCaseSupplier.TypedData(new BytesRef("YeAr"), DataType.TEXT, "chrono"),
```

Comment:
I would refactor those text/keyword tests with a for+`DataType.stringTypes()`, and add a case asking for nanos as the chrono field